apps/web/app/api/ably-auth/route.ts

```diff
@@ -0,0 +1,36 @@
+import { isProd } from "@/constants/contracts";
+import { CHANGE_EVENT_CHANNEL_NAME } from "@/globals";
+import Ably from "ably";
+import { NextResponse, NextRequest } from "next/server";
+import { HttpRequestError } from "viem";
+import { HttpCodes } from "../utils";
+
+export async function POST(req: NextRequest, res: NextResponse) {
+  // Used for linter that fails
+  if (!process.env.NEXT_ABLY_API_KEY) {
+    console.error("NEXT_ABLY_API_KEY env must be");
+    return NextResponse.json({
+      status: HttpCodes.serverError,
+      message: "No auth",
+    });
+  }
+
+  const ably = new Ably.Rest({ key: process.env.NEXT_ABLY_API_KEY });
+
+  try {
+    const tokenRequestData = {
+      capability: JSON.stringify({
+        [CHANGE_EVENT_CHANNEL_NAME]: ["publish", "subscribe", "presence"],
+      }),
+    };
+    const tokenDetails = await ably.auth.createTokenRequest(tokenRequestData);
+    return NextResponse.json(tokenDetails);
+  } catch (error) {
+    console.error(error);
+    return NextResponse.json({
+      status: 500,
+      message: "Failed to generate token",
+      error: isProd ? undefined : error,
```

Comment:
special reason for preview in vercel with errors dont return error on it?

Reply:
Not sure to understand :S

Reply:
it means in main dont show the reason for the errors? ah okay

Reply:
knowing the websocket provider could be an extra attack vector (like every link between app conponents)